flocker/node/script.py

```diff
@@ -259,9 +274,18 @@ def get_service(self, reactor, options):
 
         :return: The ``AgentLoopService`` instance.
         """
-        configuration = agent_config_from_file(path=options[u'agent-config'])
+        try:
+            agent_config = options[u'agent-config']
+            configuration = yaml.safe_load(agent_config.getContent())
+        except IOError:
+            raise ConfigurationError(
+                "Configuration file does not exist at '{}'.".format(
+                    agent_config.path))
```

Comment:
This is an anti-pattern: handle an exception with multiple possible causes and convert it into a more general exception that doesn't differentiate between those causes (with a twist: claim it is one specific cause without knowing for sure).

Please file an issue for fixing this and link to it from here.